selene_sdk/utils/config_utils.py

```diff
@@ -132,6 +137,124 @@ def initialize_model(model_configs, train=True, lr=None):
     return model, criterion
 
 
+def create_data_source(configs, output_dir=None, load_train_val=True, load_test=True):
+    """
+    Construct data source(s) used in training/evaluation.
+
+    Parameters
+    ----------
+    configs : dict or object
+        The loaded configurations from a YAML file.
+    output_dir : str or None
+        The path to the directory where all outputs will be saved.
+        If None, this means that an `output_dir` was not specified
+        in the top-level configuration keys. `output_dir` must be
+        specified in each class's individual configuration wherever
+        it is required.
+    load_train_val: bool
+        Return training and validation data loaders.
+        Only works when `"dataset" in configs`.
+    load_test: bool
+        Return test data loader. Only works when `"dataset" in configs`.
+
+    Returns
+    -------
+    model : Sampler or \
+        dataloaders : tuple(torch.utils.data.DataLoader)
+        Returns either a single data sampler specified in configs or
+        a tuple of data loaders according `load_train_val` and `load_test`,
+        which are not mutually exclusive.
+    """
+    if "sampler" in configs:
+        sampler_info = configs["sampler"]
+        if output_dir is not None:
+            sampler_info.bind(output_dir=output_dir)
+        sampler = instantiate(sampler_info)
+        return sampler
+    if "dataset" in configs:
+        dataset_info = configs["dataset"]
+        train_intervals = []
+        val_intervals = []
+        test_intervals = []
+        with open(dataset_info["sampling_intervals_path"]) as f:
+            for line in f:
+                chrom, start, end = line.rstrip().split("\t")[:3]
```

Comment:
what if there are more than 3 fields in the line?

Reply:
1. It doesn't matter, I'm only interested in interval regions specified in the file, not any additional info. And `[:3]` should take care of that anyway
2. By design of this file (generated by [this script](https://github.com/sberbank-ai-lab/DeepCT/blob/master/data/generate_target_files.py)) it shouldn't contain anything but intervals